drivers/generic_gpio_common.c

```diff
@@ -0,0 +1,444 @@
+/*
```

Comment:
Add the skeleton GPL header with your copyrights into the new source files, please.